### DISCUSSION TITLE
Font Flickering on First load in Next js with Fountsource and Material Ui?

### DISCUSSION THREAD

**Discussion by siamahnaf** [vercel/next.js#37010] [Help]
I am building one `Next.js` applications. For styling I use `Material-ui`. I use [Inter](https://fontsource.org/fonts/inter) font from FontSource.

I am facing one flickering problem in this site-
![K5XJO](https://user-images.githubusercontent.com/78312010/169033559-856d28ec-41b2-42c5-bfba-d40d3eb72867.gif)

I can't understand why I am seeing this type of flicker?

This is the live url- https://hardjourney.vercel.app/login

Here is my code example-

I install `@fontsource/inter` this and just call it in my `_app.tsx`-

```
//Fonts
import "@fontsource/inter/300.css";
import "@fontsource/inter/400.css";
import "@fontsource/inter/500.css";
import "@fontsource/inter/600.css";
import "@fontsource/inter/700.css";
```

And then I add this in Material-ui theme-

```
const theme = createTheme({
    typography: {
        fontFamily: "Inter"
    }
});
export default theme;
```

Font working but problem is that, the site load first time without font and then load fonts. What is the problem? Where I have to change in my applications? Please help me.

I test it without `Material-ui`. Without `material-ui` I am facing same problem. Without Material-ui. I tried two ways. First of all by adding `Fontsource`

`import "@fontsource/inter";`

And then `Global.css`-
```
body {
  padding: 0;
  margin: 0;
  font-family: "Inter", sans-serif;
}
```

And the second way I try to add Link in Next Documents-

```
import Document, { Html, Head, Main, NextScript } from 'next/document'

class MyDocument extends Document {
    render() {
        return (
            <Html>
                <Head>
                    <link
                        href="https://fonts.googleapis.com/css2?family=Inter&display=optional"
                        rel="stylesheet"
                    />
                </Head>
                <body>
                    <Main />
                    <NextScript />
                </body>
            </Html>
        )
    }
}

export default MyDocument
```

But not working. Can anyone help me.

**Comment by icyJoseph** [comment]
This is called `FOUT`, flash of unstyled text, and it is somewhat of a difficult thing to "just solve", one has to read and understand the trade-offs associated. There might even be problems with `FOIT`, flash of invisible text, that one might introduce unintentionally, see this [article](https://www.zachleat.com/web/mitt-romney-webfont-problem/), for something that happened back in 2015.

Long story short, the browser needs to fetch the font from the web, and that's an asynchronous process that might take very long, or not even complete. 

Now that being said, you gotta do `font-display: block`, if you don't want FOUT. This, has consequences, there'll be no text for the time the font is being downloaded, and tools like lighthouse penalise this. You should google about FOUT handling techniques, and understand all of the implications clearly.

~~I am not sure exactly where in the Material UI API you have to introduce the `font-display` declaration. I'll do a bit of reading.~~

Perhaps this: https://github.com/mui/material-ui/issues/14646#issuecomment-466769908, ?

I think in your case, using the second method to load the fonts you need with display block might be the easiest. Right now you load all families of Inter, you can narrow down to the ones you need, and slap the display block at the end.

Another solution for FOUT, is to find a good fallback font, that users typically have, Arial and such, but that's also not straightforward.
